rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/terminal/LSPTerminalREPL.java

```diff
@@ -133,26 +129,31 @@ protected Evaluator buildEvaluator(Reader input, PrintWriter stdout, PrintWriter
             stdout.println("Rascal-lsp Version: " + getRascalLspVersion());
             new StandardTextWriter(true).write(pcfg.asConstructor(), stdout);
 
+
+            // the interpreter must find the Rascal sources of util::LanguageServer etc.
+            pcfg = pcfg.addSourceLoc(JarURIResolver.jarify(lspJar));
+
+            // the interpreter must load the Java parts for calling util::IDEServices and registerLanguage
+            pcfg = pcfg.addLibLoc(lspJar);
+
+            stdout.println("Rascal " + RascalManifest.getRascalVersionNumber());
+            stdout.println("Rascal-lsp " + getRascalLspVersion());
+            stdout.println(pcfg);
+            services.registerDiagnostics(pcfg.getMessages());
+
+
             for (IValue srcPath : pcfg.getSrcs()) {
                 ISourceLocation path = (ISourceLocation)srcPath;
                 evaluator.addRascalSearchPath(path);
                 reg.watch(path, true, d -> sourceLocationChanged(path, d));
             }
 
-            ClassLoader cl = new SourceLocationClassLoader(
-                pcfg.getClassloaders()
-                    .append(URIUtil.correctLocation("lib", "rascal",""))
-                    .append(URIUtil.correctLocation("lib", "rascal-lsp",""))
-                    .append(URIUtil.correctLocation("target", projectName, "")),
-                ClassLoader.getSystemClassLoader()
-            );
-
+            ClassLoader cl = new SourceLocationClassLoader(pcfg.getLibsAndTarget(), ClassLoader.getSystemClassLoader());
```

Comment:
Pcfg.messages must be printed or registered as diagnostics or both.

Reply:
They are, a few lines higher (see `services.registerDiagnostics(pcfg.getMessages());`).

I've now also added a call to `Messages.write`